src/Symfony/Component/Cache/SimpleCache.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheException as Psr6CacheException;
+use Psr\SimpleCache\CacheInterface;
+use Psr\SimpleCache\CacheException as SimpleCacheException;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SimpleCache implements CacheInterface
+{
+    private $pool;
+    private $createCacheItem;
+
+    public function __construct(CacheItemPoolInterface $pool)
+    {
+        $this->pool = $pool;
+
+        if ($pool instanceof Adapter\AdapterInterface) {
+            $this->createCacheItem = \Closure::bind(
+                function ($key, $value) {
+                    CacheItem::validateKey($key);
+                    $item = new CacheItem();
+                    $item->key = $key;
+                    $item->value = $value;
+
+                    return $item;
+                },
+                null,
+                CacheItem::class
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        try {
+            $item = $this->pool->getItem($key);
+        } catch (SimpleCacheException $e) {
+            throw $e;
+        } catch (Psr6CacheException $e) {
+            throw new InvalidArgumentException($e->getMessage());
```

Comment:
I would link the original exception as previous exception, to keep the original stack trace (same everywhere of course)